metrics/metrics-api/src/main/java/org/apache/james/metrics/api/NoopMetricFactory.java

```diff
@@ -42,6 +42,11 @@ public void add(int value) {
         @Override
         public void remove(int value) {
         }
+
+        @Override
+        public long getCount() {
+            return 0;
```

Comment:
I would have a counter in the NoopMetric because it doesn't publish anywhere but if now the contract of a Metric is that we can rely on event count, this one must implement it

Reply:
IMO it's NoopMetric, not ThrowingMetric, I don't want it breaks our tests using this class.

Reply:
> I would have a counter in the NoopMetric because it doesn't publish anywhere but if now the contract of a Metric is that we can rely on event count, this one must implement it

We already have the `RecordingMetricFactory` for serious usage in tests. I don't even want to have test for that NoopMetric because finally it's just a dummy object to satisfy the compiler

Reply:
I disagree with the idea of letting think the caller that it works if it do not work.

We can choose either @mbaechler  proposal or mine, but I disagree hidding this problem and pretend it is ok...

Reply:
Yet another solution would be to have a separate API for metrics that do support reading:

 - Create a `ReadableMetric` interface, that extends `Metric` API
 - Create a `RedableMetricFactory` that extends `MetricFactory`, and returns `ReadableMetric` instead of `Metric`

Your health check would thus require a `RedableMetricFactory`.

NoopMetricFactory will not implement `RedableMetricFactory` hence no complicated implementation that don't have its place in a noop factory, no throwing, no 'fake answer'. What do you think?

Reply:
> Another solution would be to get rid of Noop metrics entirely...

After having a conversation with Benoit, I will do this removal in another PR, because I consider the `NoopMetric` is already providing fake behaviors in `increment(), decrement(), ...` It will be replaced by a working `RecordingMetric`

Reply:
the issue is addressed by another PR https://github.com/linagora/james-project/pull/3000